app/models/shipit/task_definition.rb

```diff
@@ -40,6 +40,14 @@ def render_title(env)
       "This task (title: #{@title}) cannot be shown due to an incorrect variable name. Check your shipit.yml file"
     end
 
+    def override_variables(params)
```

Comment:
But if you go with the controller + model, then I think it would make more sense to call this `override_defaults` and set `@default` rather than a new `@override` variable.